### PR TITLE
Generates plink files for GRM

### PR DESCRIPTION
**Pull request by evagradovich**
## Overview

- grm_plink_input parameter is now optional, creation of LD-pruned, genome-wide plink files is handled in the pipeline before SAIGE step
- sampleFile parameter in SAIGE step and its hard-coded input is removed. It is not needed if vcfs are used as input for SAIGE >0.38 and we are using version 0.39. 
- parameterises phenotype column name - previously hard-coded to PHE

## Test

Without --grm_plink_input:

https://cloudos.lifebit.ai/app/jobs/617c1efd2fc32701e672d61f

With --grm_plink_input for reproducibility of older jobs:

https://cloudos.lifebit.ai/app/jobs/617c1f142fc32701e672d8c4

### CODE REVIEW

File: bin/concat_chroms.R [hunk 1]
```diff
@@ -86,8 +86,7 @@ suppressWarnings(suppressMessages(library(snakecase)))
 # Facilitates testing and protects from wh-spaces, irregular chars
 
 # required
-output_tag         <- snakecase::to_snake_case(as.character(args$output_tag))
-
+output_tag         <- as.character(args$output_tag)
```
Comment (cgpu) [inline bin/concat_chroms.R:89]: retain  `snakecase::to_snake_case` as it strips whitespaces that can break file paths in S3

File: nextflow.config [hunk 2]
```diff
@@ -25,24 +25,31 @@ params {
   q_filter = '0.005:minor'
   miss_test_p_threshold = '1e-5'
   variant_missingness = '0.01'
-  thres_HWE = '1e-5'
-  plink_keep_pheno = "s3://lifebit-featured-datasets/projects/gel/gel-gwas/testdata/nodata"
+  hwe_threshold = '1e-5'
+  hwe_test = 'midp'
+  high_LD_long_range_regions = "s3://manu-data-public/pca-gwas-data/high-LD-regions-hg38-GRCh38.txt"
```
Comment (cgpu) [inline nextflow.config:30]: This was initially found here:
https://github.com/meyer-lab-cshl/plinkQC/blob/master/inst/extdata/high-LD-regions-hg38-GRCh38.txt

Comment (cgpu) [inline nextflow.config:30]: Replace the s3 bucket with s3://lifebit-featured-datasets

File: nextflow.config [hunk 3]
```diff
@@ -25,24 +25,31 @@ params {
...
+
+  ld_window_size = 50
+  ld_step_size = 10
+  ld_r2_threshold = 0.1
```
Comment (cgpu) [inline nextflow.config:34]: Adds docs and extra set of LD pruning steps

File: bin/concat_chroms.R (hunk 1 above)
Comment (evagradovich) [reply bin/concat_chroms.R]: Snake case re-added

File: main.nf [hunk 4]
```diff
@@ -333,33 +400,34 @@ publishDir "${params.outdir}/MultiQC/", mode: 'copy'
...
 file(gwas_cat) from ch_gwas_cat
+val(output_tag) from ch_output_tag_report
 
 output:
 file "multiqc_report.html" into ch_report_outputs
 set file("*png"), file("*ipynb"), file("*csv") into ch_report_outputs_all
 
 script:
-
+final_output_tag = output_tag.toString().toLowerCase().replaceAll(" ", "_")
```
Comment (cgpu) [inline main.nf:410]: @evagradovich thanks for diy-ing snakecase from R in Nextflow 🙏 🚀 !

**Review by cgpu — APPROVED** [review]
@evagradovich great additions, looks good to go for this iteration to me 👍

**Pull request merged**